src/MusicTheory/ScaleClass.elm

```diff
@@ -0,0 +1,728 @@
+module MusicTheory.ScaleClass exposing
+    ( ScaleClass
+    , aeolian
+    , altered
+    , alteredDoubleFlat7
+    , arabian
+    , augmented
+    , balinese
+    , blues
+    , byzantine
+    , chinese
+    , diminishedHalfToneWholeTone
+    , diminishedWholeToneHalfTone
+    , dorian
+    , dorianFlat9
+    , dorianSharp11
+    , doubleHarmonicMinor
+    , egyptian
+    , eightToneSpanish
+    , enigmatic
+    , harmonicMinor
+    , hirajoshi
+    , hungarianMajor
+    , ichikosucho
+    , ionian
+    , ionianSharp5
+    , kumoi
+    , leadingWholeTone
+    , locrian
+    , locrianNatural13
+    , locrianNatural9
+    , lydian
+    , lydianAugmented
+    , lydianDiminished
+    , lydianDominant
+    , lydianMinor
+    , lydianSharp9
+    , major
+    , majorPentatonic
+    , melodicMinor
+    , minor
+    , minorPentatonic
+    , mixolydian
+    , mixolydianFlat13
+    , mixolydianFlat9Flat13
+    , neapolitan
+    , neapolitanMajor
+    , neapolitanMinor
+    , pelog
+    , persian
+    , phrygian
+    , prometheus
+    , prometheusNeopolitan
+    , purviTheta
+    , sixToneSymmetrical
+    , todiTheta
+    , wholeTone
+    )
+
+import MusicTheory.Internals.ScaleClass as Internal
+import MusicTheory.Interval as Interval exposing (Interval)
+import MusicTheory.PitchClass as PitchClass exposing (PitchClass)
+
+
+type alias ScaleClass =
+    Internal.ScaleClass
+
+
+
+-- Modes of major
```

Comment:
Maybe it would be nice to include human readable names. But we can do this later.

E.g.:

"Diminished (wholetone, halftone)" etc.

If a client has to define that themselves that is a ton of work.

Reply:
I agree, `toString` would be nice for these. I think we can merge without it though.